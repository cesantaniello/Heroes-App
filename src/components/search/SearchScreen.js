import { useForm } from "../../hooks/useForm";

export const SearchScreen = () => {

    const [formValues, handleInputChange] = useForm({
        searchText: '',
    });

    const { searchText } = formValues;

    const handleSearch = (e) => {
        e.preventDefault();
        console.log(searchText)
    }

    return (
        <>
            <h1>Search a hero</h1>
            <hr />

            <div className="row">
                <div className="col-5">
                    <h4>Search</h4>
                    <hr />

                    <form 
                        className="d-grid gap-2"
                        onSubmit={handleSearch}>
                        <input 
                            type="text"
                            placeholder="Search a hero"
                            className="form-control"
                            name="searchText"
                            autoComplete="off"
                            value={searchText}
                            onChange={handleInputChange}
                        />

                        <button 
                            className="btn btn-primary mt-3"
                            type="submit">
                            Search...
                        </button>
                    </form>
                </div>
            </div>
        </>
    )
}