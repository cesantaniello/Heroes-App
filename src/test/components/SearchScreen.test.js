import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import { SearchScreen } from "../../components/search/SearchScreen";


describe('Tests on <SearchScreen />', () => {
    test('Should show correctly with default values', () => {
        const wrapper = mount(
        <MemoryRouter initialEntries={['/search']}>
            <SearchScreen />
        </MemoryRouter>
        );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.alert-info').text().trim()).toBe('Search a hero')
    })
})