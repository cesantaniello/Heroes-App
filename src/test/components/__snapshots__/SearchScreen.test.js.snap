// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <SearchScreen /> Should show correctly with default values 1`] = `
Array [
  <h1>
    Search a hero
  </h1>,
  <hr />,
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Search
      </h4>
      <hr />
      <form
        className="d-grid gap-2"
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Search a hero"
          type="text"
          value=""
        />
        <button
          className="btn btn-primary mt-3"
          type="submit"
        >
          Search...
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        className="alert alert-info"
      >
        Search a hero
      </div>
    </div>
  </div>,
]
`;
